packages/compiler/src/template/pipeline/src/phases/attribute_extraction.ts

```
@@ -64,13 +77,29 @@ function populateElementAttributes(view: ViewCompilation, compatibility: boolean
         break;
 
       case ir.OpKind.Property:
-      case ir.OpKind.InterpolateProperty:
         ownerOp = lookupElement(elements, op.target);
         ir.assertIsElementAttributes(ownerOp.attributes);
+        removeIfExpressionIsEmpty(op, op.expression);
+        ownerOp.attributes.add(op.bindingKind, op.name, null);
+        break;
 
+      case ir.OpKind.InterpolateProperty:
+        ownerOp = lookupElement(elements, op.target);
+        ir.assertIsElementAttributes(ownerOp.attributes);
         ownerOp.attributes.add(op.bindingKind, op.name, null);
         break;
 
+      case ir.OpKind.StyleProp:
+        ownerOp = lookupElement(elements, op.target);
+        ir.assertIsElementAttributes(ownerOp.attributes);
+
+        // The old compiler treated empty style bindings as regular bindings for the purpose of
+        // directive matching. That behavior is incorrect, but we emulate it in compatibility mode.
+        if (removeIfExpressionIsEmpty(op, op.expression) && compatibility) {
```

As an alternative option, we could just throw an error for a style binding to an empty expression. @crisbeto pointed out that we do this for empty binding to listeners.

---

IMO I would just throw the error as part of the ingestion step, but I think either approach is fine enough.

---

I could definitely do that, but we would lose the ability to have the `compatibility` mode where we do what the old compiler did. Its an edge case that I doubt too many people are really hitting. @alxhub curious if you have an opinion here